.eslintrc.local.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+const fs = require('fs')
+
+const pkgJson = JSON.parse(fs.readFileSync('./package.json', 'utf-8'))
+
+module.exports = {
+  plugins: ['import'],
+  rules: {
+    'import/no-extraneous-dependencies': [
```

Comment:
this may not be required. checking

Reply:
It's already in the base ruleset https://github.com/npm/eslint-config/blob/9e87b58d9e049004c802239cab1fabbca67caf74/lib/index.js#L213

Reply:
Ah yep. I think this config is useful though? The default would allow importing from `devDependencies`

Reply:
The file is already imported as a plugin, so the extra dependency isn't needed.  I think a single new `rule` is all we need here with an `allow` of `./test` and `map.js`.

Reply:
cf https://github.com/npm/cli/blob/3d5bbcc86c484ef0b6c04ee0236c1072d36915d6/.eslintrc.local.json#L3 or https://github.com/npm/cacache/blob/de12400af2ab452f1e75a15edaf7ea51b294667b/.eslintrc.local.js#L4

Reply:
Cool. Will remove the dependency. Feels a bit strange though not having it because without that the plugin is only there as a side effect of `@npmcli/template-oss` bringing it in

Reply:
Done that. If at some point the plugin is removed from `@npmcli/eslint-config` it looks like eslint will complain that the rule is missing

Reply:
I still think this config needs to be removed, it's part of our core linting already. We're only concerned about the nodejs modules rule.

Reply:
But if someone imports something that’s not a dependency (or peer dep), that would cause errors for users?

I think the default config would not catch those

Reply:
It will currently.  We have that rule enabled on all of the oss repos we manage through template-oss via the npm eslint config.


```sh
~/D/n/n/b/main (main|✔) $ npm run lint

> semver@7.5.0 lint
> eslint "**/*.js"


/Users/wraithgar/Development/npm/node-semver/branches/main/bin/test.js
  1:7   error  'nopt' is assigned a value but never used                                             no-unused-vars
  1:14  error  'nopt' should be listed in the project's dependencies. Run 'npm i -S nopt' to add it  import/no-extraneous-dependencies
```

Reply:
It's enabled, but the default config wouldn't catch an import from `devDependencies`

## Without config
<img width="779" alt="image" src="https://user-images.githubusercontent.com/3259993/236539965-a80f97cd-15de-4e61-8e0d-7041e618de07.png">

## With config in the pr
<img width="927" alt="image" src="https://user-images.githubusercontent.com/3259993/236540084-bf50a374-8050-4fdd-82bb-cd07d7f87192.png">

Reply:
Then this needs to be addressed in template-oss, that's not a rule that we'd want to only apply to a single one of our over 80 repos.

Reply:
Opened https://github.com/npm/template-oss/pull/316

Reply:
@wraithgar so given this doesn't follow the standard structure https://github.com/npm/eslint-config/pull/67 doesn't have any effect here, so is the custom config in this PR good to go?

Reply:
Yeah we'll need a custom config but this one looks to be a bit different in approach than the one we landed on in eslint-config.  We should pattern it after the one there but instead of `lib` and `bin` it should be `bin`, `classes`, `functions`, `internal`, and `ranges` (and now we begin to see why we standardized on `lib`).

The way this PR currently does it is by completely bypassing the `no-extraneous-dependencies` rule for the test files, which is a regression.

Reply:
Yep. done!